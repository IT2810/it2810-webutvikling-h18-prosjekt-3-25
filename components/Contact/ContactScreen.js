import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Avatar, Icon } from 'react-native-elements';

export default class Contact extends React.Component {
    //Her settes tittelen som brukes i navigasjonsbaren
    static navigationOptions = {
        title: "Contact",
    }

    render() {
        return (
            <View style={styles.contact}>
                <View style={styles.picture}>
                    <View style={styles.backButton}>
                        <Icon
                            name="arrow-back"
                            color="white"
                        />
                    </View>

                    <View style={styles.avatar}>
                        <Avatar
                            xlarge
                            title={this.props.navigation.getParam("name").split(" ").map(x => x[0]).join("")}
                            overlayContainerStyle={{backgroundColor: "#abcccc"}}
                        />
                    </View>
                    <View style={styles.picturePadding}>
                        {/*This is needed to be able to center the avatar*/}
                    </View>

                </View>
                <View style={styles.texts}>

                    {/* this.props.navigation.getParam() tilsvarer this.props. uten bruk av navigasjonsbiblioteket */}
                    <View style={styles.infoLine}>
                        <Icon name="person"/>
                        <Text style={styles.text}>{this.props.navigation.getParam("name")}</Text>
                    </View>
                    <View style={styles.infoLine}>
                        <Icon name="phone" />
                        <Text style={styles.text}>{this.props.navigation.getParam("number")}</Text>
                    </View>
                    <View style={styles.infoLine}>
                        <Icon name="email" />
                        <Text style={styles.text}>{this.props.navigation.getParam("email")}</Text>
                    </View>

                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({

    contact: {
        flex: 1,
    },

    picture: {
        backgroundColor: "#abcccc",
        flex: 1,
        flexDirection: "row",
    },

    backButton: {
        flex: 1,
        alignItems: "flex-start",
    },

    avatar: {
        alignItems: "center",
        justifyContent: "center",
    },

    picturePadding: {
        flex: 1,
    },

    texts: {
        flex: 2,
    },

    infoLine: {
        flexDirection: "row",
        paddingLeft: 10,
        borderBottomWidth: 1,
        borderBottomColor: "#aaa",
    },

    text: {
        fontSize: 18,
        paddingBottom: 15,
        paddingTop: 15,
        paddingLeft: 20,
    },
});
