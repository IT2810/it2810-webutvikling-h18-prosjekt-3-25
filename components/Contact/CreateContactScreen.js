import React from 'react';
import { StyleSheet, Text, View, KeyboardAvoidingView } from 'react-native';
import { Icon, FormLabel, FormInput, FormValidationMessage } from 'react-native-elements';

export default class CreateContact extends React.Component {
    //Her settes tittelen som brukes i navigasjonsbaren
    static navigationOptions = {
        title: "Create Contact"
    }

    constructor(props) {
        super(props);
        this.state = {
            name: "",
            number: "",
            email: "",
        }

        this.handleOnPress = this.handleOnPress.bind(this);
    };

    //Lager et kontakt-objekt ut fra informasjonen som er lagret i state
    //Setter så state til opprinnelig form igjen, for så å navigere til kontaklisten
    handleOnPress() {
        const contact = {
            name: this.state.name,
            number: this.state.number,
            email: this.state.email,
        }

        //Dette tilsvarer this.props.onPress(contact) uten bruk av navigasjonsbiblioteket
        this.props.navigation.getParam("onPress")(contact);
        this.setState(state => ({
            name: "",
            number: "",
            email: "",
        }));
        this.props.navigation.goBack();
    }

    render() {
        return (
            // KeyboardAvoidingView gjør at skjermbildet tilpasser seg det digitale
            // tastaturet
            <KeyboardAvoidingView style={styles.createContactContainer} behaivor="padding" >
                <View style={styles.icon}>
                    <Icon
                        name="person-add"
                        size={50}
                        onPress={this.handleOnPress}
                    />
                </View>
                <View>
                    {/* Oppdaterer hvert av feltene i state etterhvert som teksten endrer seg */}
                    <FormLabel>Name</FormLabel>
                    <FormInput onChangeText={(text) => this.setState(state => ({name: text}))} />

                    <FormLabel>Number</FormLabel>
                    <FormInput onChangeText={(text) => this.setState(state => ({number: text}))} keyboardType="numeric"/>

                    <FormLabel>Email</FormLabel>
                    <FormInput onChangeText={(text) => this.setState(state => ({email: text}))} keyboardType="email-address"/>
                </View>
            </KeyboardAvoidingView>
        );
    }
}

const styles = StyleSheet.create({
    createContactContainer: {
        flex: 1,
    },

    icon: {
        padding: 10,
        alignItems: "flex-end",
    },
});
