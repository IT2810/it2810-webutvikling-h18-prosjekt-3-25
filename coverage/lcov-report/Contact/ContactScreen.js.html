<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Contact/ContactScreen.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\prettify.css" />
    <link rel="stylesheet" href="..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\index.html">All files</a> / <a href="index.html">Contact</a> ContactScreen.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/3</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Avatar, Icon } from 'react-native-elements';
&nbsp;
export default class Contact extends React.Component {
    //Her settes tittelen som brukes i navigasjonsbaren
    static navigationOptions = {
        title: "Contact",
    }
&nbsp;
<span class="fstat-no" title="function not covered" >    re</span>nder() {
<span class="cstat-no" title="statement not covered" >        return (</span>
            &lt;View style={styles.contact}&gt;
                &lt;View style={styles.picture}&gt;
                    &lt;Avatar
                        xlarge
                        title={this.props.navigation.getParam("name").split(" ").map(<span class="fstat-no" title="function not covered" >x </span>=&gt; <span class="cstat-no" title="statement not covered" >x[0])</span>.join("")}
                        overlayContainerStyle={{backgroundColor: "#abcccc"}}
                    /&gt;
                &lt;/View&gt;
                {/* this.props.navigation.getParam() tilsvarer this.props. uten bruk av navigasjonsbiblioteket */}
                &lt;View style={styles.infoLine}&gt;
                    &lt;Icon name="person"/&gt;
                    &lt;Text style={styles.text}&gt;{this.props.navigation.getParam("name")}&lt;/Text&gt;
                &lt;/View&gt;
                &lt;View style={styles.infoLine}&gt;
                    &lt;Icon name="phone" /&gt;
                    &lt;Text style={styles.text}&gt;{this.props.navigation.getParam("number")}&lt;/Text&gt;
                &lt;/View&gt;
                &lt;View style={styles.infoLine}&gt;
                    &lt;Icon name="email" /&gt;
                    &lt;Text style={styles.text}&gt;{this.props.navigation.getParam("email")}&lt;/Text&gt;
                &lt;/View&gt;
            &lt;/View&gt;
        );
    }
}
&nbsp;
const styles = StyleSheet.create({
&nbsp;
    picture: {
        backgroundColor: "#abcccc",
        alignItems: "center",
    },
&nbsp;
    infoLine: {
        flexDirection: "row",
        paddingLeft: 10,
        borderBottomWidth: 1,
        borderBottomColor: "#aaa",
    },
&nbsp;
    text: {
        fontSize: 18,
        paddingBottom: 15,
        paddingTop: 15,
        paddingLeft: 20,
    },
});
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Oct 10 2018 13:09:25 GMT+0200 (Vest-Europa (sommertid))
</div>
</div>
<script src="..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\sorter.js"></script>
</body>
</html>
