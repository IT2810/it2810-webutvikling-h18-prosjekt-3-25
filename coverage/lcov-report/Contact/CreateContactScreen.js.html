<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Contact/CreateContactScreen.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\prettify.css" />
    <link rel="stylesheet" href="..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\index.html">All files</a> / <a href="index.html">Contact</a> CreateContactScreen.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">31.25% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">20% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">41.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/12</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { StyleSheet, Text, View, KeyboardAvoidingView } from 'react-native';
import { Icon, FormLabel, FormInput, FormValidationMessage } from 'react-native-elements';
&nbsp;
export default class CreateContact extends React.Component {
    //Her settes tittelen som brukes i navigasjonsbaren
    static navigationOptions = {
        title: "Create Contact"
    }
&nbsp;
    constructor(props) {
        super(props);
        this.state = {
            name: "",
            number: "",
            email: "",
        }
&nbsp;
        this.handleOnPress = this.handleOnPress.bind(this);
    };
&nbsp;
    //Lager et kontakt-objekt ut fra informasjonen som er lagret i state
    //Setter så state til opprinnelig form igjen, for så å navigere til kontaklisten
<span class="fstat-no" title="function not covered" >    ha</span>ndleOnPress() {
        const contact = <span class="cstat-no" title="statement not covered" >{</span>
            name: this.state.name,
            number: this.state.number,
            email: this.state.email,
        }
&nbsp;
        //Dette tilsvarer this.props.onPress(contact) uten bruk av navigasjonsbiblioteket
<span class="cstat-no" title="statement not covered" >        this.props.navigation.getParam("onPress")(contact);</span>
<span class="cstat-no" title="statement not covered" >        this.setState(<span class="fstat-no" title="function not covered" >st</span>ate =&gt; (<span class="cstat-no" title="statement not covered" >{</span></span>
            name: "",
            number: "",
            email: "",
        }));
<span class="cstat-no" title="statement not covered" >        this.props.navigation.goBack();</span>
    }
&nbsp;
    render() {
        return (
            &lt;KeyboardAvoidingView style={styles.createContactContainer} behaivor="padding" &gt;
                &lt;View style={styles.icon}&gt;
                    &lt;Icon
                        name="person-add"
                        size={50}
                        onPress={this.handleOnPress}
                    /&gt;
                &lt;/View&gt;
                &lt;View&gt;
                    {/* Oppdaterer hvert av feltene i state etterhvert som teksten endrer seg */}
                    &lt;FormLabel&gt;Name&lt;/FormLabel&gt;
                    &lt;FormInput onChangeText={<span class="fstat-no" title="function not covered" >(t</span>ext) =&gt; <span class="cstat-no" title="statement not covered" >this.setState(<span class="fstat-no" title="function not covered" >st</span>ate =&gt; (<span class="cstat-no" title="statement not covered" >{name: text})</span>)}</span> /&gt;
&nbsp;
                    &lt;FormLabel&gt;Number&lt;/FormLabel&gt;
                    &lt;FormInput onChangeText={<span class="fstat-no" title="function not covered" >(t</span>ext) =&gt; <span class="cstat-no" title="statement not covered" >this.setState(<span class="fstat-no" title="function not covered" >st</span>ate =&gt; (<span class="cstat-no" title="statement not covered" >{number: text})</span>)}</span> keyboardType="numeric"/&gt;
&nbsp;
                    &lt;FormLabel&gt;Email&lt;/FormLabel&gt;
                    &lt;FormInput onChangeText={<span class="fstat-no" title="function not covered" >(t</span>ext) =&gt; <span class="cstat-no" title="statement not covered" >this.setState(<span class="fstat-no" title="function not covered" >st</span>ate =&gt; (<span class="cstat-no" title="statement not covered" >{email: text})</span>)}</span> keyboardType="email-address"/&gt;
                &lt;/View&gt;
            &lt;/KeyboardAvoidingView&gt;
        );
    }
}
&nbsp;
const styles = StyleSheet.create({
    createContactContainer: {
        flex: 1,
    },
&nbsp;
    icon: {
        padding: 10,
        alignItems: "flex-end",
    },
});
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Oct 10 2018 13:09:25 GMT+0200 (Vest-Europa (sommertid))
</div>
</div>
<script src="..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\sorter.js"></script>
</body>
</html>
