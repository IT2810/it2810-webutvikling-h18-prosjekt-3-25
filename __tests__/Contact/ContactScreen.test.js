import React from 'react';
import Contact from '../../components/Contact/ContactScreen.js';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

// Trengs for Ã¥ la Contact mota props, siden disse blir sendt med ract-navigation.
const navigation = {getParam: jest.fn(() => {
    return "test Person";
})}

it("renders correctly", () => {
    const tree = renderer.create(<Contact navigation={navigation} />).toJSON();
    expect(tree).toMatchSnapshot();
})
