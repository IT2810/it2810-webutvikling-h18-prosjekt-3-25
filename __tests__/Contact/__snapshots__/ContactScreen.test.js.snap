// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#abcccc",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 150,
            "paddingBottom": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "width": 150,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "rgba(0,0,0,0.2)",
              "bottom": 0,
              "flex": 1,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
            Object {
              "backgroundColor": "#abcccc",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "backgroundColor": "rgba(0,0,0,0)",
                  "color": "#ffffff",
                  "fontSize": 75,
                  "textAlign": "center",
                },
                undefined,
              ],
            ]
          }
        >
          tP
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "paddingLeft": 10,
      }
    }
  >
    <View
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 24,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "paddingBottom": 15,
          "paddingLeft": 20,
          "paddingTop": 15,
        }
      }
    >
      test Person
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "paddingLeft": 10,
      }
    }
  >
    <View
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 24,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "paddingBottom": 15,
          "paddingLeft": 20,
          "paddingTop": 15,
        }
      }
    >
      test Person
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "paddingLeft": 10,
      }
    }
  >
    <View
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 24,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "paddingBottom": 15,
          "paddingLeft": 20,
          "paddingTop": 15,
        }
      }
    >
      test Person
    </Text>
  </View>
</View>
`;
